src/momento-signer.ts

```diff
@@ -0,0 +1,213 @@
+import {InvalidArgumentError} from '.';
+import {importJWK, JWK, JWTHeaderParameters, KeyLike, SignJWT} from 'jose';
+
+export enum CacheOperation {
+  Get = 'GET',
+  Set = 'SET',
+}
+
+export interface signingRequest {
+  /**
+   * @param {string} cacheName - Name of the cache the signed token should be allowed to operate on.
+   */
+  cacheName: string;
+  /**
+   * @param {number} expiryEpochSeconds - Timestamp in epoch seconds for
+   * when the signed token should expire.
+   */
+  expiryEpochSeconds: number;
+  /**
+   * @param {string} [cacheKey] - The cache key the signed token is allowed to
+   * access. If not present, auth token is allowed to be used for any key.
+   */
+  cacheKey?: string;
+  /**
+   * @param {CacheOperation} [cacheOperation] - The operation the signed token
+   * should be valid for. If not present the signed token is allowed to be used
+   * for any operation.
+   */
+  cacheOperation?: CacheOperation;
+  /**
+   * @param {number} [ttlSeconds] - The time to live for cache items that this
+   * signed token can set. If not present the signed token is allowed to be
+   * used for set operations with any ttl.
```

Comment:
This is more of a comment to keep in mind for our service in the future: it's probably not very helpful to have an exact TTL constraint. We should really allow the signer to specify a `max_ttl` constraint, and requests must only set TTLs _up to that value_, instead of enforcing the exact TTL value on the backend service (again, not relevant to this PR, so just a drive-by comment).

Reply:
Yeah that makes a lot of sense to me, but agreed its outside the scope of this PR. I can open an issue for it though.